src/graphql/node/control.rs

```diff
@@ -39,172 +38,222 @@ impl NodeControlMutation {
         }
     }
 
-    // TODO: Apply node issue #251
     #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
         .or(RoleGuard::new(Role::SecurityAdministrator))")]
-    async fn apply_node(&self, ctx: &Context<'_>, id: ID) -> Result<ApplyResult> {
+    async fn apply_node(&self, ctx: &Context<'_>, id: ID) -> Result<ApplyNodeResponse> {
         let i = id.as_str().parse::<u32>().map_err(|_| "invalid ID")?;
-        let agents = ctx.data::<BoxedAgentManager>()?;
 
-        let (node, _invalid_agents) = {
+        let (node, invalid_agents) = {
             let store = crate::graphql::get_store(ctx).await?;
             let node_map = store.node_map();
             node_map
                 .get_by_id(i)?
-                .ok_or_else(|| async_graphql::Error::new(format!("Node with ID {i} not found",)))?
+                .ok_or_else(|| async_graphql::Error::new(format!("Node with ID {i} not found")))?
         };
 
-        if node.name_draft.is_none() && node.profile_draft.is_none() {
-            return Err("There is nothing to apply.".into());
+        if !invalid_agents.is_empty() {
+            return Err(async_graphql::Error::new(format!(
+                "Node {i} cannot be applied due to invalid agents: {invalid_agents:?}"
+            )));
         }
 
-        let config_setted_modules = send_set_config_requests(agents, &node).await;
-        let success_modules = if let Ok(config_setted_modules) = config_setted_modules {
-            update_node(ctx, i, node.clone(), &config_setted_modules).await?;
+        if !valid_node_apply(&node) {
+            return Err("Node is not valid for apply".into());
+        }
 
-            if let Some(customer_id) = should_broadcast_customer_change(&node) {
-                broadcast_customer_change(customer_id, ctx).await?;
-            }
-            config_setted_modules
-        } else {
-            return Err("Failed to apply node profile".into());
-        };
+        let giganto_draft = node
+            .giganto
+            .as_ref()
+            .and_then(|g| g.draft.as_ref().map(ToString::to_string));
 
-        Ok(ApplyResult {
-            id,
-            success_modules,
-        })
+        let apply_scope = node_apply_scope(&node);
+
+        if let Some(ref target_agents) = apply_scope.agents {
+            let agent_manager = ctx.data::<BoxedAgentManager>()?;
+            notify_agents(
+                agent_manager,
+                &target_agents.updates,
+                &target_agents.disables,
+            )
+            .await?;
+
+            info!(
+                "[{}] Node ID {i} - Node's agents are notified to apply.\n{:?}",
```

Comment:
The term 'apply' means any updates will be applied to the node. And, each agent
will be updated with a new configuration. Therefore, I suggest the message like
"The node are notified to update its agents."

Reply:
Or, "Node's agents are notified to be updated."

Reply:
Done